grails-app/views/stockMovement/_events.gsp

```diff
@@ -0,0 +1,137 @@
+                    </td>
+                    <td>
+                        ${historyItem?.date}
+                    </td>
+                    <td>
+                        ${historyItem?.location}
+                    </td>
+                    <td>
+                        ${historyItem?.referenceDocument?.identifier}
+                    </td>
+                    <td>
+                        ${historyItem?.createdBy}
+                    </td>
+                    <td>
+                        ${historyItem?.comment}
+                    </td>
+                </tr>
+            </g:each>
+            </tbody>
+        </table>
+    </g:if>
+    <g:else>
+        <div class="fade center empty"><g:message code="default.noEvents.label" default="No events"/></div>
+    </g:else>
+</section>
+
+<section class="box">
+    <h2>
+        <img src="${resource(dir: 'images/icons/silk', file: 'add.png')}" alt="event" style="vertical-align: middle"/>
+        <label>
+            <g:message code="default.addCustomEvent.label" default="Add Custom Event"/>
+        </label>
+    </h2>
+    <g:form controller="shipment" action="saveEvent" method="POST">
+        <g:hiddenField name="shipmentId" value="${shipmentId}" />
+        <g:hiddenField name="createdBy.id" value="${session.user.id}" />
+        <table>
+            <tbody>
+            %{-- Event type picker --}%
+            <tr class="prop">
+                <td valign="top" class="name">
+                    <label>
+                        <g:message code="eventType.label" default="Event type"/>
+                    </label>
+                </td>
+                <td valign="top" class="value ${hasErrors(bean: eventInstance, field: 'eventType', 'errors')}">
+                <g:select id="eventType.id"
+                          name="eventType.id"
+                          optionKey="id"
+                          noSelection="['': g.message(code: 'default.selectOne.label')]"
+                          from='${EventType.listCustomEventTypes()}'
+                          optionValue="${{ format.metadata(obj: it) }}"
+                          required="${true}"
+                          class="chzn-select-deselect">
+                </g:select>
+                </td>
+            </tr>
+            %{-- Location picker --}%
+            <tr class="prop">
+                <td valign="top" class="name">
+                    <label>
+                        <g:message code="default.location.label" default="Location" />
+                    </label>
+                </td>
+                <td valign="top" class="value ${hasErrors(bean: eventInstance, field: 'location', 'errors')}">
+                    <g:select id="eventLocation.id"
+                              name="eventLocation.id"
+                              noSelection="['': warehouse.message(code: 'default.selectOne.label')]"
+                              from='${Location.list()}'
```

Comment:
Ooof, that `Location.list()` might be an overkill. On obdev3 this gives us `51646` options with all locations in the system (even inactive). What should be a potential location to choose here? Depots, Wards, Internal Locations? Can we somehow narrow this or do an async select?

Reply:
Additional question - should this be required? On the old event tab, it looks like it is preselected with origin and cannot be cleared.

Reply:
> AND the Location field selects the value “null” by default